spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceProperties.java

```diff
@@ -520,39 +520,39 @@ public void setProperties(Map<String, String> properties) {
 
 	}
 
-	static class DataSourceBeanCreationException extends BeanCreationException {
+	class DataSourceBeanCreationException extends BeanCreationException {
```

Comment:
I'm not sure whether I should remove `static` modifier. Maybe it would be even better to move this class outside of `DataSourceProperties`?